changelogutils/github.go

```diff
@@ -0,0 +1,68 @@
+package changelogutils
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-github/github"
+	"github.com/pkg/errors"
+	"github.com/solo-io/go-utils/githubutils"
+	"github.com/solo-io/go-utils/versionutils"
+)
+
+func GetProposedTagFromGit(ctx context.Context, client *github.Client, owner, repo, ref string) (string, error) {
+	latestTag, err := GetLatestTag(ctx, owner, repo)
+	if err != nil {
+		return "", err
+	}
+	files, err := githubutils.GetFilesFromGit(ctx, client, owner, repo, ref, ChangelogDirectory)
+	if err != nil {
+		return "", nil
+	}
+
+	proposedVersion := ""
+	for _, file := range files {
+		if file.GetType() == githubutils.CONTENT_TYPE_DIRECTORY {
+			version := file.GetName()
+			if !versionutils.MatchesRegex(version) {
+				return "", errors.Errorf("Directory name %s is not valid, must be of the form 'vX.Y.Z'", version)
+			}
+
+			proposedVersion, err = validVersion(version, proposedVersion, latestTag)
+		}
+	}
+	if proposedVersion == "" {
+		return "", errors.Errorf("No version greater than %s found", latestTag)
+	}
+	return proposedVersion, err
+}
+
+func ValidateProposedChangelogTag(ctx context.Context, client *github.Client, owner, repo, ref string) (bool, error) {
+	proposedTag, err := GetProposedTagFromGit(ctx, client, owner, repo, ref)
```

Comment:
The flow here is a bit unexpected, the test for this file first calls getProposed, then calls validateProposed which internally calls getProposed again. I'd suggest we slightly change it:
* GetProposedTag -> takes a ref, figures out which, if any, tag / directory is the proposed release tag (proposed = higher semver version than current). returns error if there are 0 or multiple proposed
* Validate -> takes a tag and the current ref, checks the directory contents: a) makes sure they are valid and parseable, b) makes sure number has been properly incremented based on whether there is a breaking change